### PR TITLE
Full BiasCorrection for OpenVINO backend

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Changes

- Added OpenVINO backend support for the BiasCorrection algorithm

### Reason for changes

- Extension of the set of algorithms for the backend

### Related tickets

- 98621

### CODE REVIEW

**Comment by alexsu52** [comment]
@andrey-churkin please, review

**Comment by alexsu52** [comment]
@andrey-churkin please, review

File: nncf/experimental/openvino_native/graph/model_transformer.py [hunk 1]
```diff
@@ -231,8 +231,11 @@ def _apply_bias_correction_transformations(self, transformations: List[OVBiasCor
 
             node = self.name_to_node_mapping[node_name]
             node_inputs = [port.get_node() for port in node.output(0).get_target_inputs()]
-            node_with_bias = node_inputs[0]
-            assert node_with_bias.get_type_name() == 'Add'
+            assert any(node.get_type_name() == 'Add' for node in node_inputs)
```
Comment (andrey-churkin) [inline nncf/experimental/openvino_native/graph/model_transformer.py:234]: Does `len(node_inputs)` always equal 1 for `node_with_bias`?

File: nncf/quantization/algorithms/bias_correction/backend.py [hunk 2]
```diff
@@ -57,7 +58,9 @@ def quantizer_types(self):
 
     @staticmethod
     @abstractmethod
-    def target_point(target_type: TargetType, target_node_name: str, port_id: str = None) -> TargetPoint:
+    def target_point(target_type: TargetType,
+                     target_node_name: str,
+                     port_id: str) -> TargetPoint:
```
Comment (andrey-churkin) [inline nncf/quantization/algorithms/bias_correction/backend.py:63]:
```suggestion
                     port_id: int) -> TargetPoint:
```

File: nncf/quantization/algorithms/bias_correction/onnx_backend.py [hunk 3]
```diff
@@ -88,45 +99,35 @@ def mean_statistic_collector(reduction_shape: ReductionShape,
...
+
+    @staticmethod
+    def get_output_name(model: onnx.ModelProto, node_name: str) -> List[str]:
         onnx_graph = ONNXGraph(model)
         node = onnx_graph.get_node_by_name(node_name)
-        return node.output
+        return node.output[0]
 
     @staticmethod
     def extract_model(model: onnx.ModelProto,
```
Comment (andrey-churkin) [inline nncf/quantization/algorithms/bias_correction/onnx_backend.py:133]: Could you please move this logic to `ONNXModelTransformer` and update FBC algorithm (`get_input_output_names()` method should be removed from backend)?

Comment (andrey-churkin) [reply nncf/quantization/algorithms/bias_correction/onnx_backend.py:133]: Could you please check maybe it is possible to remove `ONNXExtendedLayerAttributes` class?

File: nncf/experimental/openvino_native/graph/model_transformer.py [hunk 4]
```diff
@@ -231,8 +231,11 @@ def _apply_bias_correction_transformations(self, transformations: List[OVBiasCor
...
+
+            for node_input in node_inputs:
+                if node_input.get_type_name() == 'Add':
+                    node_with_bias = node_input
```
Comment (andrey-churkin) [inline nncf/experimental/openvino_native/graph/model_transformer.py:238]:
```suggestion
                    add_node= node_input
```

File: nncf/experimental/openvino_native/quantization/algorithms/bias_correction/openvino_backend.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    def get_activation_port_ids_for_bias_node(node: NNCFNode) -> Tuple[int, int]:
+        return 0, 0
+
+    @staticmethod
+    def get_bias_value(node: NNCFNode, model: ov.Model, nncf_graph: NNCFGraph) -> np.ndarray:
+        return get_bias_value(node, nncf_graph, model)
+
+    @staticmethod
+    def get_bias_port_id(node: NNCFNode, model: ov.Model) -> int:
+        return node.layer_attributes.weight_port_id
```
Comment (andrey-churkin) [inline nncf/experimental/openvino_native/quantization/algorithms/bias_correction/openvino_backend.py:113]: This method is not used anymore. Could you please remove it?

File: nncf/quantization/algorithms/bias_correction/algorithm.py [hunk 6]
```diff
@@ -431,36 +441,48 @@ def output_filter_func(point):
 
     def get_statistic_points(self, model: TModel) -> StatisticPointsContainer:
         self._set_backend_entity(model)
-        nncf_graph = NNCFGraphFactory.create(model) if self.nncf_graph is None else self.nncf_graph
+        model_copy = self._remove_fq_from_inputs(deepcopy(model))
```
Comment (andrey-churkin) [inline nncf/quantization/algorithms/bias_correction/algorithm.py:444]: Should we use this method here?
https://github.com/openvinotoolkit/nncf/blob/7a1de11e3b9c9d641b7f44faa271ac059700f252/nncf/common/utils/backend.py#L90-L101
@l-bat This issue was fixed?

File: nncf/quantization/algorithms/bias_correction/algorithm.py [hunk 7]
```diff
@@ -472,16 +494,49 @@ def _get_biased_after_input_nodes(self, nncf_graph: NNCFGraph, model_inputs: Lis
...
+        """
+        Returns 0- activation node, but not quanitzers.
+
+        :param node: NNCFNode instance.
+        :param nncf_graph: NNCFGraph instance.
+        :return: NNCFNode activation node.
+        """
+        activation_input_port = 0
+        previous_node = nncf_graph.get_input_edges(node)[activation_input_port].from_node
+        return previous_node
```
Comment (andrey-churkin) [inline nncf/quantization/algorithms/bias_correction/algorithm.py:523]: I don't see the reason why this method is necessary. The docstring is not very clear to me.
My opinion is it is better to use
```python
node = nncf_graph.get_input_edges(node)[0].from_node
```
and add comments for this line to give more information if it is needed.

File: nncf/quantization/algorithms/bias_correction/onnx_backend.py (hunk 3 above)
Comment (KodiaqQ) [reply nncf/quantization/algorithms/bias_correction/onnx_backend.py]: `ONNXExtendedLayerAttributes` can't be removed due to their's usage in the `node_utils.py`.

File: nncf/quantization/algorithms/bias_correction/algorithm.py (hunk 6 above)
Comment (l-bat) [reply nncf/quantization/algorithms/bias_correction/algorithm.py:444]: The issue was fixed om master branch. We should wait for release of `openvino-dev`.

**Comment by andrey-churkin** [comment]
I have reviewed it. It looks good to me. The PR is being tested now. Guys,  if you would like to review this PR, please add your comments. It will be merged after confirmation if there are no other comments.

**Review by andrey-churkin — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]
LGTM

**Pull request merged**